File: 242-ValidAnagram/validAnagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        used = [0] * 26
        for i in range(len(s)):
            used[ord(s[i]) - 97] += 1
            used[ord(t[i]) - 97] -= 1

        for i in used:
            if i != 0:
                return False
        return True

File: 242-ValidAnagram/validAnagram.java
class Solution {
  public boolean isAnagram(String s, String t) {
    int[] used = new int[26];
    for (char c : s.toCharArray()) {
      used[c - 'a']++;
    }
    for (char c : t.toCharArray()) {
      used[c - 'a']--;
    }

    for (int i = 0; i < used.length; i++) {
      if (used[i] != 0) return false;
    }

    return true;
  }
}